File: CS/T983248_MVC/Controllers/HomeController.cs
using DevExpress.Web;
using DevExpress.Web.Mvc;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;
using System.Web.Mvc;
using T983248_MVC.Models;

namespace T983248_MVC.Controllers
{
    public class HomeController : Controller
    {
        public ActionResult Index()
        {
            return View(new UserModel());
        }

        [HttpPost]
        public ActionResult Index(UserModel modelDTO)
        {
            string fileName = string.Empty;
            if (ModelState.IsValid)
            {
                if (modelDTO.Attachment.Length > 0 && modelDTO.Attachment[0].ContentLength > 0)
                {
                    fileName = string.Format("~/Content/Files/{0}", modelDTO.Attachment[0].FileName);
                    modelDTO.Attachment[0].SaveAs(Server.MapPath(fileName));
                }
            }
            SavedModel model = new SavedModel();
            model.UserName = modelDTO.UserName;
            model.FileUrl = fileName;
            return View("Complete", model);
        }
    }
}

File: VB/T983248_VB/Controllers/HomeController.vb
Public Class HomeController
    Inherits System.Web.Mvc.Controller

    Public Function Index() As ActionResult
        Return View(New UserModel())
    End Function

    <HttpPost>
    Public Function Index(ByVal modelDTO As UserModel) As ActionResult
        Dim fileName As String = String.Empty

        If ModelState.IsValid Then

            If modelDTO.Attachment.Length > 0 AndAlso modelDTO.Attachment(0).ContentLength > 0 Then
                fileName = String.Format("~/Content/Files/{0}", modelDTO.Attachment(0).FileName)
                modelDTO.Attachment(0).SaveAs(Server.MapPath(fileName))
            End If
        End If

        Dim model As SavedModel = New SavedModel()
        model.UserName = modelDTO.UserName
        model.FileUrl = fileName
        Return View("Complete", model)
    End Function
End Class